scripts/ensure_exports_are_documented.py

```diff
@@ -52,10 +84,18 @@ def get_names_from_all_list() -> list[str]:
 
 def ensure_exports_are_documented() -> bool:
     success = True
-    for name in get_names_from_all_list():
-        if not is_documented(name):
-            print(f"{name} is not documented")
-            success = False
+    used_deprecations = set()
+    for package_dir in ("globus_sdk", "globus_sdk/scopes"):
+        for name in get_names_from_all_list(package_dir):
+            if name in DEPRECATED_NAMES:
+                used_deprecations.add(name)
+                continue
+            if not is_documented(name):
+                print(f"{name} is not documented")
```

Comment:
I know this isn't new, but this error message could use a bit of work.

I'd ask that we convey:
  1. What is required in terms of documentation (an `.. auto<x>` method)
  2. Ideally some sort of trace to this file (not sure what the best format for that would be? maybe just a file name?). In the case of an exception it can be difficult to trace down and understand exactly what's being evaluated and why.